File: backend/src/main/kotlin/com/example/backend/models/Caisse.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "caisse")
class Caisse {
  @Id
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id")
  var user: com.example.backend.models.Employe? = null

  @Column(name = "ouvertureCaisse")
  var ouvertureCaisse: String? = null

  @Column(name = "fermetureCaisse")
  var fermetureCaisse: String? = null

  @Column(name = "dateOuvert")
  var dateOuvert: LocalDateTime? = null

  @Column(name = "dateFerme")
  var dateFerme: LocalDateTime? = null

  @Column(name = "session", length = 32)
  var session: String? = null

  @Column(name = "fondCaisseOuvert")
  var fondCaisseOuvert: Double? = 0.0

  @Column(name = "fondCaisseFerme")
  var fondCaisseFerme: Double? = 0.0

  @Column(name = "etat", length = 16)
  var etat: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/caisse.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.core.db import Base

class Caisse(Base):
  __tablename__ = "caisse"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, index=True)
  user_id = Column(Integer, ForeignKey("employe.id"), nullable=False)
  user = relationship("Employe", lazy="joined")
  ouverture_caisse = Column(String)
  fermeture_caisse = Column(String)
  date_ouvert = Column(DateTime)
  date_ferme = Column(DateTime)
  session = Column(String(32))
  fond_caisse_ouvert = Column(Float, default=0.0)
  fond_caisse_ferme = Column(Float, default=0.0)
  etat = Column(String(16))
  supprimer = Column(Integer, default=0)
